web/src/services/language.ts

```diff
@@ -79,3 +79,56 @@ export const setLanguage = async (l: string) => {
 export const init = async () => {
     await setLanguage(current.value.code)
 }
+
+
+export const currentAudioLanguage = computed(() => {
+    localStorage.setItem("audioLanguage", localStorage.getItem("audioLanguage") ?? "en");
+    var code = localStorage.getItem("audioLanguage");
+    return languages.value.find(
+        (l) => l.code === code
+    ) as Language 
+})
+
+export const subtitleLanguage = computed(() => {
+    localStorage.setItem("subtitleLanguage", localStorage.getItem("subtitleLanguage") ?? "en");
+    var code = localStorage.getItem("subtitleLanguage");
+    return languages.value.find(
+        (l) => l.code === code
+    ) as Language 
+})
+
+export const setAudioLanguage = (l: string) => {
+    localStorage.setItem("audioLanguage", l);
+    languages.value = getLanguages(current.value.code)
+}
+
+export const setSubtitleLanguage = (l: string) => {
+    localStorage.setItem("subtitleLanguage", l);
+    languages.value = getLanguages(current.value.code)
+}
+
+export const getNestedLangComputed = (title: string) => {
+    switch (title) {
+        case 'Site Language':
+            return current
+        case 'Audio Language':
+            return currentAudioLanguage
+        case 'Subtitle Language':
+            return subtitleLanguage
+        default:
+            return;
+    }
+}
+
+export const setNestedLangMethod = (title: string, code: string) => {
```

Comment:
You should probably not send "Title" here.

Use properly typed strings: `"content" | "audio" | "subtitle"`